lib/timers.js

```diff
@@ -409,6 +393,26 @@ function processImmediate() {
   }
 }
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw) {
+      if (!L.isEmpty(queue)) {
+        // Handle any remaining on next tick, assuming we're still
+        // alive to do so.
+        while (!L.isEmpty(immediateQueue)) {
+          L.append(queue, L.shift(immediateQueue));
+        }
+        immediateQueue = queue;
+        process.nextTick(processImmediate);
+      }
```

Comment:
I'm not sure what you asking here. If we `return` inside the `finally` block, then the exception won't be thrown, which I don't think would be ideal behavior.